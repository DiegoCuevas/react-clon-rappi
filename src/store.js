import { createStore, applyMiddleware, compose } from "redux";
import reducer from "./reducer";
import thunk from "redux-thunk";
import logger from "redux-logger";
import offlineSync from "./midlewares/offline-sync";

const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

if (localStorage.getItem("version") !== process.env.REACT_APP_VERSION) {
  localStorage.removeItem("state");
}
const preloadedState = JSON.parse(localStorage.getItem("state") || "{}");

const store = createStore(
  reducer,
  // preloadedState,
  // composer(applyMiddleware(thunk, logger, offlineSync))
  composer(applyMiddleware(thunk, logger))
);

export default store;
