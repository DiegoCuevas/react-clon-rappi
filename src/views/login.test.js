import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import store from "../store";
import Login from "./login";

test("Login component", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Login />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
