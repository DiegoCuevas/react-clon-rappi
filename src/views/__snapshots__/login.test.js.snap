// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component 1`] = `
<DocumentFragment>
  .emotion-7 {
  max-width: 500px;
  margin: 0 auto;
}

.emotion-0 {
  margin: 3rem 0;
  font-size: 2rem;
  text-align: center;
}

.emotion-6 {
  border-radius: .5em;
  box-sizing: border-box;
  border: 1px solid #fff;
  padding: 15px;
  width: 100%;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.25s;
  transition-duration: 0.25s;
  margin-top: 3rem;
}

.emotion-6:hover {
  cursor: pointer;
  border: 1px solid #e5edef;
  box-shadow: -1px 2px 5px 0px rgba(0,0,0,.12);
}

.emotion-1 {
  display: block;
  margin: .5rem 0;
}

.emotion-2 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  width: 100%;
}

.emotion-2:focus {
  outline: none;
  border-color: #444444;
}

.emotion-5 {
  background-color: black;
  border: 1px solid black;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: .8rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-5:hover {
  background-color: white;
  color: black;
}

<section
    class="emotion-7"
  >
    <h1
      class="emotion-0"
    >
      Login to the Application
    </h1>
    <div
      class="emotion-6"
    >
      <form>
        <label
          class="emotion-1"
          for="email"
        >
          Email
        </label>
        <input
          class="emotion-2"
          id="email"
          name="email"
          type="email"
          value="josh@delivery.pe"
        />
        <label
          class="emotion-1"
          for="password"
        >
          Password
        </label>
        <input
          class="emotion-2"
          id="password"
          name="password"
          type="password"
          value="123456"
        />
        <br />
        <button
          class="emotion-5"
        >
          Log In
        </button>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
